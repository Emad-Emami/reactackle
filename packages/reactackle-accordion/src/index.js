import Accordion from './Accordion';
import AccordionItem from './AccordionItem/AccordionItem';

export { Accordion, AccordionItem };
