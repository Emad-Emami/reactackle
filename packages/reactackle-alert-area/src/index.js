import AlertArea from './AlertArea';

export { AlertArea };
