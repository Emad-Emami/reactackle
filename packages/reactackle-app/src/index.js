import App from './App';

export { App };
export * from './BottomRegion';
export * from './TopRegion';
export * from './MainRegion';
