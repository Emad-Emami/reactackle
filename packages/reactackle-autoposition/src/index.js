import AutoPosition, { recalculateAllAutoPosition } from './AutoPosition';

export { AutoPosition, recalculateAllAutoPosition };
