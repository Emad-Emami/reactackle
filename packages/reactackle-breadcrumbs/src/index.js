import Breadcrumbs from './Breadcrumbs';

export { Breadcrumbs };
export * from './BreadcrumbsItem';
