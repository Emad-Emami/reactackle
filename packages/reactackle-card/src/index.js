import Card from './Card';

export { Card };
export * from './CardActions';
export * from './CardAreaMain';
export * from './CardAreaSide';
export * from './CardContent';
export * from './CardHeaderPrimary';
export * from './CardHeaderSecondary';
export * from './CardMedia';
