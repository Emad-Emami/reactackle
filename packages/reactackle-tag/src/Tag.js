import React from 'react';
import PropTypes from 'prop-types';
import { IconCross } from 'reactackle-icons';
import { withTooltip } from 'reactackle-tooltip';
import { noop, registerDefaultComponentTheme } from 'reactackle-core';

import {
  componentTheme,
  TagStyled,
  IconStyled,
  ContentStyled,
  TextStyled,
  RemoveIconStyled,
} from './styles';

registerDefaultComponentTheme('tag', componentTheme);

/*
 * Combined with tooltip
 */

const propTypes = {
  /*
   * Bounded tag has max width, long text may be cut
   */
  bounded: PropTypes.bool,
  /*
   * Define icon component
   */
  icon: PropTypes.element,
  /*
   * Tag text
   */
  text: PropTypes.string,
  /*
   * Tooltip text
   */
  tooltipText: PropTypes.string,
  /*
   * Redefines background color
   */
  bgColor: PropTypes.string,
  /*
   * Redefines text color
   */
  textColor: PropTypes.string,
  /*
   * Redefines icon color
   */
  iconColor: PropTypes.string,
  /*
   * Adds remove button
   */
  removable: PropTypes.bool,
  /*
   * Remove Icon
   */
  iconRemove: PropTypes.element,
  /*
   * Function to be called in remove button click
   */
  onRemove: PropTypes.func,
  /**
   * @ignore
   */
  toggleTooltip: PropTypes.func,
  /**
   * @ignore
   */
  showTooltip: PropTypes.func,
  /**
   * @ignore
   */
  hideTooltip: PropTypes.func,
};

const defaultProps = {
  bounded: false,
  icon: null,
  text: '',
  tooltipText: '',
  bgColor: '',
  textColor: '',
  iconColor: '',
  removable: false,
  iconRemove: <IconCross size="custom" color="currentColor" />,
  onRemove: noop,
  toggleTooltip: noop,
  showTooltip: noop,
  hideTooltip: noop,
};

const TagComponent = ({
  bounded,
  removable,
  icon,
  iconRemove,
  text,
  tooltipText,
  bgColor,
  textColor,
  iconColor,
  children,
  ...props
}) => {
  const tooltip = tooltipText && <props.Tooltip text={tooltipText} />;
  
  const textElement = text && (
    <TextStyled isBounded={bounded} title={text}>
      {text}
    </TextStyled>
  );

  const removeButton = removable && (
    <RemoveIconStyled onClick={props.onRemove}>
      {iconRemove}
    </RemoveIconStyled>
  );

  const iconElement = icon && (
    <IconStyled iconColor={iconColor}>
      {icon}
    </IconStyled>
  );
  
  const wrapperProps = {
    onClick: props.toggleTooltip,
    onFocus: props.showTooltip,
    onBlur: props.hideTooltip,
    onMouseEnter: props.showTooltip,
    onMouseLeave: props.hideTooltip,
  };
  
  return (
    <TagStyled
      {...wrapperProps}
      bgColorCustom={bgColor}
      textColorCustom={textColor}
    >
      {iconElement}
      <ContentStyled
        hasIcon={!!icon}
        isBounded={bounded}
        isRemovable={removable}
      >
        {textElement}
        {children}
      </ContentStyled>
      {removeButton}
      {tooltip}
    </TagStyled>
  );
};

TagComponent.propTypes = propTypes;
TagComponent.defaultProps = defaultProps;
TagComponent.displayName = 'Tag';

export default withTooltip(TagComponent, true);
