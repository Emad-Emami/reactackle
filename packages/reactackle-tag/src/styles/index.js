import componentTheme from './theme';

export { componentTheme };

export * from './TagStyled';
export * from './IconStyled';
export * from './ContentStyled';
export * from './TextStyled';
export * from './RemoveIconStyled';
