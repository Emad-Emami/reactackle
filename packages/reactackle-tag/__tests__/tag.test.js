import React from 'react';
import renderer from 'react-test-renderer';
import { IconSvg } from 'reactackle-icons';
import { Tag } from '../src';

jest.mock('react-dom');

describe('<Tag />', () => {
  it('renders correctly with default props', () => {
    const tree = renderer.create(
      <Tag />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with bgColor prop set to red', () => {
    const tree = renderer.create(
      <Tag bgColor='red' />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with bounded prop set to true', () => {
    const tree = renderer.create(
      <Tag bounded />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with icon prop set to <IconSvg />', () => {
    const tree = renderer.create(
      <Tag icon={<IconSvg />} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with icon prop set to IconSvg component', () => {
    const tree = renderer.create(
      <Tag icon={IconSvg} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with iconColor prop set to red', () => {
    const tree = renderer.create(
      <Tag icon={IconSvg} iconColor='red' />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with iconRemove prop set to <IconSvg />', () => {
    const tree = renderer.create(
      <Tag iconRemove={<IconSvg />} removable />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with text prop set to some text', () => {
    const tree = renderer.create(
      <Tag text='some text' />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with textColor prop set to red', () => {
    const tree = renderer.create(
      <Tag text='some text' textColor='red' />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with tooltipText prop set to String', () => {
    const tree = renderer.create(
      <Tag tooltipText='some text' />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
