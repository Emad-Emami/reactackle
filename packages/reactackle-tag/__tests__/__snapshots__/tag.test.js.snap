// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tag /> renders correctly with bgColor prop set to red 1`] = `
<div
  className="TagStyled-dxhDsf gbJpRz"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI bPGsom"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with bounded prop set to true 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI giaxec"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with default props 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI bPGsom"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with icon prop set to <IconSvg /> 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="IconStyled-bPbExM kwdUPZ"
  >
    <div
      className="IconSvgStyled-cegYAl iSGIaX"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
        />
      </svg>
    </div>
  </span>
  <span
    className="ContentStyled-dbeIsI ckvEZg"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with icon prop set to IconSvg component 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="IconStyled-bPbExM kwdUPZ"
  >
    <div
      className="IconSvgStyled-cegYAl iSGIaX"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
        />
      </svg>
    </div>
  </span>
  <span
    className="ContentStyled-dbeIsI ckvEZg"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with iconColor prop set to red 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="IconStyled-bPbExM loIaJY"
  >
    <div
      className="IconSvgStyled-cegYAl iSGIaX"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
        />
      </svg>
    </div>
  </span>
  <span
    className="ContentStyled-dbeIsI ckvEZg"
  >
    
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with iconRemove prop set to <IconSvg /> 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI eMazQd"
  >
    
  </span>
  <button
    className="RemoveIconStyled-guscKJ cROeMa"
    onClick={[Function]}
  >
    <div
      className="IconSvgStyled-cegYAl iSGIaX"
      color=""
      rotate={0}
      size="normal"
    >
      <svg
        preserveAspectRatio="xMidYMid meet"
        viewBox="0 0 24 24"
      >
        <path
          d="M19,5.5c0-0.1,0-0.1,0-0.2c-0.1-0.1-0.1-0.2-0.3-0.3c-0.1,0-0.1,0-0.2,0h-10C8.4,5,8.4,5,8.3,5c-0.1,0-0.1,0.1-0.2,0.1l-3,3 c0,0-0.1,0.1-0.1,0.2c0,0.1,0,0.1,0,0.2v10C5,18.8,5.2,19,5.5,19h10c0.1,0,0.1,0,0.2,0c0.1,0,0.1-0.1,0.2-0.1l3-3 c0,0,0.1-0.1,0.1-0.2c0-0.1,0-0.1,0-0.2V5.5z M16,16h1.3L16,17.3V16z M18,15h-2V8.7l2-2V15z M9,9h6v6H9V9z M8,8H6.7L8,6.7V8z M9,6 h8.3l-2,2H9V6z M6,9h2v6.5C8,15.8,8.2,16,8.5,16H15v2H6V9z"
        />
      </svg>
    </div>
  </button>
  
</div>
`;

exports[`<Tag /> renders correctly with text prop set to some text 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI bPGsom"
  >
    <span
      className="TextStyled-ezUzxo byCMDZ"
      title="some text"
    >
      some text
    </span>
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with textColor prop set to red 1`] = `
<div
  className="TagStyled-dxhDsf gvVpVk"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI bPGsom"
  >
    <span
      className="TextStyled-ezUzxo byCMDZ"
      title="some text"
    >
      some text
    </span>
  </span>
  
</div>
`;

exports[`<Tag /> renders correctly with tooltipText prop set to String 1`] = `
<div
  className="TagStyled-dxhDsf cCxqOx"
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <span
    className="ContentStyled-dbeIsI bPGsom"
  >
    
  </span>
</div>
`;
